x-pack/plugins/apm/server/routes/metrics/by_agent/serverless/compute_usage.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import {
+  kqlQuery,
+  rangeQuery,
+  termQuery,
+} from '@kbn/observability-plugin/server';
+import { euiLightVars as theme } from '@kbn/ui-theme';
+import {
+  FAAS_BILLED_DURATION,
+  METRICSET_NAME,
+  METRIC_SYSTEM_TOTAL_MEMORY,
+  SERVICE_NAME,
+} from '../../../../../common/elasticsearch_fieldnames';
+import { environmentQuery } from '../../../../../common/utils/environment_query';
+import { isFiniteNumber } from '../../../../../common/utils/is_finite_number';
+import { getVizColorForIndex } from '../../../../../common/viz_colors';
+import { getMetricsDateHistogramParams } from '../../../../lib/helpers/metrics';
+import { Setup } from '../../../../lib/helpers/setup_request';
+import { GenericMetricsChart } from '../../fetch_and_transform_metrics';
+import { ChartBase } from '../../types';
+
+const chartBase: ChartBase = {
+  title: i18n.translate('xpack.apm.agentMetrics.serverless.computeUsage', {
+    defaultMessage: 'Compute usage',
+  }),
+  key: 'compute_usage',
+  type: 'linemark',
+  yUnit: 'number',
+  series: {
+    computeUsage: {
+      title: i18n.translate('xpack.apm.agentMetrics.serverless.computeUsage', {
+        defaultMessage: 'Compute usage',
+      }),
+    },
+  },
+};
+
+/**
+ * To calculate the compute usage we need to multiple the "system.memory.total" by "faas.billed_duration".
+ * But the result of this calculation is in Bytes-milliseconds, as the "system.memory.total" is stored in bytes and the "faas.billed_duration" is stored in milliseconds.
+ * But to calculate the overall cost AWS uses GB-second, so we need to convert the result to this unit.
+ */
+function calculateComputeUsageGBSeconds({
+  faasBilledDuration,
+  totalMemory,
+}: {
+  faasBilledDuration?: number | null;
+  totalMemory?: number | null;
+}) {
+  if (!isFiniteNumber(faasBilledDuration) || !isFiniteNumber(totalMemory)) {
+    return 0;
+  }
+  const bytesMsResult = totalMemory * faasBilledDuration;
+  const GBSeconds = 1024 * 1024 * 1024 * 1000;
```

Comment:
```suggestion
  const GBSeconds = 1000 * 1024**3;
```

Reply:
It would be clearer for me to define something like:
```ts
const GB = 1024**3;
const totalMemoryGB = totalMemory /  GB;
const faasBilledDurationSec = faasBilledDuration * 1000;
return totalMemoryGB * faasBilledDurationSec;
```
but maybe it's just me not being used to thinking about these units.

Reply:
@AlexanderWert came up with this conversion, so I think we should be good!